gateway/shard/manager.go

```diff
@@ -206,16 +194,30 @@ func (m *Manager) ApplyError(f func(g *gateway.Gateway) error, all bool) error {
 	m.mutex.RLock()
 	defer m.mutex.RUnlock()
 
+	var errs MultiError
+
 	for _, g := range m.gateways {
 		if err := f(g); err != nil {
-			return &Error{
+			wrapperErr := &Error{
 				ShardID: shardID(g),
 				Source:  err,
 			}
+
+			if !all {
+				return wrapperErr
+			}
+
+			errs = append(errs, wrapperErr)
 		}
 	}
 
-	return nil
+	if len(errs) == 0 {
+		return nil
+	} else if len(errs) == 1 {
```

Comment:
Unnecessary `else` here. In fact, consider making it a `switch` on `len(errs)` instead.

Reply:
Why is the else unnecessary? If we just encounter a single error, we should return just that one instead of the MultiError. Also we only have two cases, so why a switch?

Reply:
The `else` is not required because the `if` contains an early return. A `switch` can contain `case 0:`, `case 1:` and `default:`.